app/controllers/staff/activity_uploads_controller.rb

```diff
@@ -13,7 +13,10 @@ def new
   end
 
   def show
-    stream_csv_download(filename: "activities.csv", headers: csv_headers)
+    @report_presenter = ReportPresenter.new(@report)
+    filename = "#{ERB::Util.url_encode(@report_presenter.financial_quarter_and_year)}-#{ERB::Util.url_encode(@report_presenter.fund.title)}-#{ERB::Util.url_encode(@report.organisation.beis_organisation_reference)}_activities_upload.csv"
+
+    stream_csv_download(filename: filename, headers: csv_headers)
```

Comment:
I'm wondering if I should have made `stream_csv_download` perform the URL-encoding of the filename -- it's really the whole thing that needs to be encoded for the context it's being put into, not individual pieces of it, and the above could be written

```rb
filename = ERB::Util.url_encode("#{@report_presenter.financial_quarter_and_year}-#{@report_presenter.fund.title}-#{@report.organisation.beis_organisation_reference}_activities_upload.csv")
```

Reply:
I did aggregate all of this together in one of the later commits, where I'm making the report presenter responsible for generating the filenames, and then it's encoding the whole string at the end.

I didn't think of messing with `stream_csv_download`, I can try that out.